packages/modules/psa/soc.py

```diff
@@ -0,0 +1,65 @@
+from typing import Union, List
+
+import logging
+
+from dataclass_utils import dataclass_from_dict
+from helpermodules.cli import run_using_positional_cli_args
+from modules.common import store
+from modules.common.abstract_device import DeviceDescriptor
+from modules.common.abstract_soc import AbstractSoc
+from modules.common.component_context import SingleComponentUpdateContext
+from modules.common.component_state import CarState
+from modules.common.fault_state import ComponentInfo
+from modules.psa import api
+from modules.psa.config import PSA, PSAConfiguration
+
+
+log = logging.getLogger("soc."+__name__)
+
+
+class Soc(AbstractSoc):
+    def __init__(self, device_config: Union[dict, PSA], vehicle: int):
+        self.config = dataclass_from_dict(PSA, device_config)
+        self.vehicle = vehicle
+        self.store = store.get_car_value_store(self.vehicle)
+        self.component_info = ComponentInfo(self.vehicle, self.config.name, "vehicle")
+
+    def update(self, charge_state: bool = False) -> None:
```

Comment:
Der Parameter `charge_state` wird nie gelesen.

Reply:
muss mir mal anschauen wie das in den anderen SoC Modulen genutzt wird, meine dass es dort ähnlich ist.

Reply:
Ach, das kommt vom `AbstractSoc`. Sorry, ich habe nichts gesagt.